app/src/main/java/com/github/jorgecastillo/kotlinandroid/domain/model/CharacterError.kt

```diff
@@ -13,7 +15,7 @@ package com.github.jorgecastillo.architecturecomponentssample.model.error
  * bring not required complexity to the architecture introducing asynchronous semantics.
  */
 sealed class CharacterError {
-  class AuthenticationError : CharacterError()
-  class NotFoundError : CharacterError()
-  class UnknownServerError : CharacterError()
+  object AuthenticationError : CharacterError()
```

Comment:
Since you are not storing any state using classes is a waste of memory, allocations and GC down the road. Both in Scala and Kotlin is considered good practice to use objects to represent cases in ADTs since they are singleton values and only one of them is enough to model the case.

They do store shared state. A singleton or object is both a type and a value. The only possible value whereas a class is just a type and every value requires a new instantiation.